### PR TITLE
Stacksafe sorter

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
Title says it all :)

### JIRA issue:
RHOL-736

### Checklist
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### CODE REVIEW

**Comment by ArturGajowy** [comment]
Supersedes #1466

File: casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala [hunk 1]
```diff
@@ -108,15 +107,18 @@ object BlockAPI {
...
         mainChain           <- getMainChainFromTip[F]
         maybeRuntimeManager <- casper.getRuntimeManager
         runtimeManager      = maybeRuntimeManager.get // This is safe. Please reluctantly accept until runtimeManager is no longer exposed.
-        sortedListeningNames = immutable.Seq(
-          listeningNames.channels.map(channelSortable.sortMatch(_).term): _*)
+        sortedListeningNames <- immutable
+                                 .Seq(
+                                   listeningNames.channels
+                                     .map(channelSortable.sortMatch[F](_).map(_.term)): _*)
+                                 .sequence
```
Comment (goral09) [inline casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: Another one `map` + `sequence` = `traverse` ;)

File: models/src/main/scala/coop/rchain/models/rholang/sort/GroundSortMatcher.scala [hunk 2]
```diff
@@ -1,56 +1,71 @@
...
+              ParMap(sortedPars.map(_.term), gm.connectiveUsed, gm.locallyFree, gm.remainder)),
+            Node(Leaf(Score.EMAP) :: sortedPars.map(_.score) ::: remainderScoreOpt.toList)
+          )
       case GByteArray(ba) =>
-        ScoredTerm(g, Node(Score.EBYTEARR, Leaf(ba.toStringUtf8)))
+        ScoredTerm(g, Node(Score.EBYTEARR, Leaf(ba.toStringUtf8))).pure[F]
       case _ => //TODO(mateusz.gorski): rethink it
-        throw new IllegalArgumentException("GroundSortMatcher passed unknown Expr instance")
+        Sync[F].raiseError(
+          new IllegalArgumentException("GwroundSortMatcher passed unknown Expr instance"))
```
Comment (goral09) [inline models/src/main/scala/coop/rchain/models/rholang/sort/GroundSortMatcher.scala]: Gwround => Ground.

**Comment by ArturGajowy** [comment]
> Just two small changes but otherwise looks good.

Would be helpful to get a :heavy_check_mark: in advance in such cases :)

**Pull request merged**